### PR TITLE
Migrate MySQL product tests to integration tests

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Follows https://trino.io/docs/current/develop/tests.html#avoid-product-tests

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by ebyhr** [comment]
The migrated test is flaky:
```
Error:    TestMySqlDynamicFiltering.testDynamicFilteringStats:59->AbstractTestQueryFramework.assertQuery:350 For query 20240717_133801_00005_ypzy9: 
 SELECT "completeddynamicfilters.totalcount" FROM jmx.current."io.trino.plugin.jdbc:name=mysql,type=dynamicfilteringstats" WHERE node = 'c474f9f1-3932-4a14-824e-79bea9a8db99'
not equal
Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):
    [0]
Expected rows (up to 100 of 1 missing rows shown, 1 rows in total):
    [1]
```

File: plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlDynamicFiltering.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                .setSystemProperty("join_reordering_strategy", "NONE")
+                .setSystemProperty("join_distribution_type", "BROADCAST")
+                .build();
+
+        assertThat(query(session, "SELECT COUNT(*) FROM nation a JOIN tpch.tiny.nation b ON a.nationkey = b.nationkey AND b.name = 'INDIA'"))
+                .matches("VALUES BIGINT '1'");
+
+        String coordinatorId = (String) computeScalar("SELECT node_id FROM system.runtime.nodes WHERE coordinator = true");
+
+        assertEventually(() -> assertThat((long) computeScalar("SELECT \"completeddynamicfilters.totalcount\" FROM jmx.current.\"io.trino.plugin.jdbc:name=mysql,type=dynamicfilteringstats\"" +
```
Comment (ebyhr) [inline plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlDynamicFiltering.java:59]: This test becomes flaky without `assertEventually` method. Perhaps, related to https://github.com/trinodb/trino/issues/2214

**Review by raunaqmorarka — APPROVED** [review]
We added this as a product test because there was some issue with relying on JMX beans in integration tests, I forget what the problem was, but if it works now this is a good change.

**Comment by raunaqmorarka** [comment]
> We added this as a product test because there was some issue with relying on JMX beans in integration tests, I forget what the problem was, but if it works now this is a good change.

Ah I see that CI failed, so maybe the problem still exists

**Comment by ebyhr** [comment]
The test works locally, but it somehow fails on CI. I will investigate the cause.

**Pull request closed without merging**